### PR TITLE
Hazard type CRUD

### PR DESCRIPTION
**Pull request by DanSid**
@sosualfred Please I am done with the CRUD

### CODE REVIEW

File: backend/src/controllers/hazardtypes.ts [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        _id: new mongoose.Types.ObjectId(),
+        name
+    });
+
+    return _hazardType
+        .save()
+        .then((hazardType) => {
+            return res.status(201).json({
+                hazardType
+            });
```
Comment (sosualfred) [inline backend/src/controllers/hazardtypes.ts:20]: Just 201 with a msg hazard created successfully is okay.

**Review by sosualfred — CHANGES_REQUESTED** [review]

File: backend/src/controllers/hazardtypes.ts (hunk 1 above)
Comment (sosualfred) [reply backend/src/controllers/hazardtypes.ts:20]: Can you use async await .then is not readable.

File: backend/src/controllers/hazardtypes.ts [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+        return res.status(200).json({
+            message: 'All the Hazard Types',  hazardTypes,
+            count: hazardTypes.length
+        });
+    } catch (error) {
+        // return res.status(500).json({
+        //     message: error.message,
+        //     error
+        // });
```
Comment (sosualfred) [inline backend/src/controllers/hazardtypes.ts:38]: Remove commented out code

File: backend/src/controllers/hazardtypes.ts [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+    const _hazardType = new HazardType({
+        _id: new mongoose.Types.ObjectId(),
+        name
+    });
+    try {
+        const hazardType = await _hazardType.save();
+        return res.status(400).json({
+            message: 'Hazard Type created successfully',     hazardType
+        });
```
Comment (sosualfred) [inline backend/src/controllers/hazardtypes.ts:18]: remove hazardType from the response.

File: backend/src/controllers/hazardtypes.ts [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+        } else {
+            return res.status(404).json({
+                message: 'Hazard Type not found'
+            });
+        }
+    } catch (error) {
+        // return res.status(500).json({
+        //     message: error.message,
+        //     error
+        // });
```
Comment (sosualfred) [inline backend/src/controllers/hazardtypes.ts:63]: Remove commented out code

File: backend/src/controllers/hazardtypes.ts [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+const getHazardTypeById = async (req: Request, res: Response, next: NextFunction) => {
+    const hazardTypeId = req.params.id;
+
+    try {
+        const hazardType = await HazardType.findById(hazardTypeId).exec();
+
+        if (hazardType) {
+            return res.status(200).json({
+              message: 'Your Specific Hazard Type',hazardType
+            });
```
Comment (sosualfred) [inline backend/src/controllers/hazardtypes.ts:53]: remove message

File: backend/src/controllers/hazardtypes.ts [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+        const hazardType = await HazardType.findById(hazardTypeId).exec();
+
+        if (hazardType) {
+            hazardType.set(req.body);
+
+            const result = await hazardType.save();
+
+            return res.status(200).json({
+                message: 'Your Specific Hazard Type',                hazardType: result
+            });
```
Comment (sosualfred) [inline backend/src/controllers/hazardtypes.ts:80]: update message to `Hazard updated successfully`, remove `hazardType`

File: backend/src/controllers/hazardtypes.ts [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+        } else {
+            return res.status(404).json({
+                message: 'HazardType not found'
+            });
+        }
+    } catch (error) {
+        // return res.status(500).json({
+        //     message: error.message,
+        //     error
+        // });
```
Comment (sosualfred) [inline backend/src/controllers/hazardtypes.ts:90]: Remove commented out code

File: backend/src/controllers/hazardtypes.ts [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+        } else {
+            return res.status(404).json({
+                message: 'Hazard Type not found'
+            });
+        }
+    } catch (error) {
+        // return res.status(500).json({
+        //     message: error.message,
+        //     error
+        // });
```
Comment (sosualfred) [inline backend/src/controllers/hazardtypes.ts:115]: Remove commented out code

**Review by sosualfred — CHANGES_REQUESTED** [review]

**Comment by DanSid** [comment]
Please @sosualfred I have made the validation and updates to the CRUD endpoints

**Comment by sosualfred** [comment]
There are still some changes you've not made. But, I'll just go ahead and merge it.

**Comment by DanSid** [comment]
Ok

**Pull request merged**